File: bot.cpp
#include <iostream>
#include <vector>

struct Movie {
  std::string title;
  int year;
  std::string genre;
};

std::vector<Movie> movies = {
  { "The Shawshank Redemption", 1994, "Drama" },
  { "The Godfather", 1972, "Crime" },
  { "The Godfather: Part II", 1974, "Crime" },
  // add more movies here
};

std::vector<Movie> searchMovies(std::string genre, int year) {
  std::vector<Movie> results;
  for (const auto& movie : movies) {
    if (movie.genre == genre && movie.year == year) {
      results.push_back(movie);
    }
  }
  return results;
}

int main() {
  auto results = searchMovies("Crime", 1972);
  for (const auto& movie : results) {
    std::cout << movie.title << std::endl;
  }
  return 0;
}

File: bot.java
import java.util.ArrayList;
import java.util.List;

public class Movie {
  private String title;
  private int year;
  private String genre;

  public Movie(String title, int year, String genre) {
    this.title = title;
    this.year = year;
    this.genre = genre;
  }

  public String getTitle() {
    return title;
  }

  public int getYear() {
    return year;
  }

  public String getGenre() {
    return genre;
  }
}

List<Movie> movies = new ArrayList<>();
movies.add(new Movie("The Shawshank Redemption", 1994, "Drama"));
movies.add(new Movie("The Godfather", 1972, "Crime"));
movies.add(new Movie("The Godfather: Part II", 1974, "Crime"));
// add more movies here

List<Movie> searchMovies(String genre, int year) {
  List<Movie> results = new ArrayList<>();
  for (Movie movie : movies) {
    if (movie.getGenre().equals(genre) && movie.getYear() == year) {
      results.add(movie);
    }
  }
  return results;
}

List<Movie> results = searchMovies("Crime", 1972);
for (Movie movie : results) {
  System.out.println(movie.getTitle());
}
